### PR TITLE
BE OAuth Configuration

### PR DESCRIPTION
**Pull request by alderwhiteford**
# Description

[Link to Ticket](insert the link to your ticket inside the parenthesis here)

Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. If they are unit
tests, provide the file name the tests are in. If they are not unit tests,
describe how you tested the change.

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have reached out to another developer to review my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by alderwhiteford** [comment]
A lot of stuff needs to be removed / moved around

File: backend/integrations/oauth/oauth.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	ResourceClient 	OAuthResourceClientInterface
+}
+
+type OAuthClientInterface interface {
+	Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) string
+	Token(requestBody models.OAuthTokenRequestBody, ctx *fasthttp.RequestCtx, userId *uuid.UUID, db *gorm.DB, jwtClient *auth.JWTClientInterface) error
+}
+
+// Return the authorization URL: 
+func (client OAuthClient) Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) (*string, error) {
```
Comment (garrettladley) [inline backend/integrations/oauth/oauth.go:29]: we should probably be using pointer receivers over value receivers so we don't have to copy the config strings around
```suggestion
func (client *OAuthClient) Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) (*string, error) {
```

File: backend/integrations/oauth/oauthResource.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+type GoogleOAuthClient struct {
+	OAuthConfig config.OAuthConfig
+}
+
+/**
+* GOOGLE OAUTH CLIENT
+**/
+func (client GoogleOAuthClient) AuthorizeURL(state string) string {
+	return (
+		client.OAuthConfig.BaseURL + "/auth?" +
```
Comment (garrettladley) [inline backend/integrations/oauth/oauthResource.go:28]: there might be a url builder utility we can use here. if not, we should probably prefer fmt.Sprintf over string concat

File: backend/integrations/oauth/transactions.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	if err := db.Create(&calendarToken).Error; err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateCalendarToken(db *gorm.DB, jwtClient auth.JWTClientInterface, userID *uuid.UUID) (*string, error) {
+	// Generate CSRF token:
+	csrfToken, err := jwtClient.GenerateToken(auth.Claims{
```
Comment (garrettladley) [inline backend/integrations/oauth/transactions.go:61]: to separate concerns on token generation and database transactions, the token should probably be generated in the service, then passed to this function as a string

File: backend/migrations/000001_init.up.sql [hunk 4]
```diff
@@ -246,4 +247,11 @@ CREATE TABLE IF NOT EXISTS verifications(
 
 CREATE UNIQUE INDEX IF NOT EXISTS uni_verifications_token ON verifications USING btree ("token");
 
+CREATE TABLE IF NOT EXISTS user_calendar_token(
```
Comment (garrettladley) [inline backend/migrations/000001_init.up.sql:250]: let's update the [id].down.sql to include a matching drop statement

File: backend/integrations/oauth/oauthResource.go
Comment (garrettladley) [inline backend/integrations/oauth/oauthResource.go:1]: nit but we should split into files for google and outlook. including the associated request bodies there as well

File: backend/entities/oauth/base/controller.go
Comment (garrettladley) [inline backend/entities/oauth/base/controller.go:1]: is there not an associated service and transactions? we should separate our concerns so integrations/oauth only handles the bespoke google/outlook requirements while we have our internal service and transaction logic in this package

**Review by garrettladley — COMMENTED** [review]
you're cooking @alderwhiteford

**Comment by garrettladley** [comment]
> A lot of stuff needs to be removed / moved around

didn't see this until after my review, sorry for redundant comments on this

**Pull request closed without merging**